continuous_integration/gpuci/build.sh

```diff
@@ -57,5 +54,8 @@ conda info
 conda config --show-sources
 conda list --show-channel-urls
 
-rapids-logger "Python py.test for dask"
-py.test $WORKSPACE -n 3 -v -m gpu --junitxml="$WORKSPACE/junit-dask.xml" --cov-config="$WORKSPACE/pyproject.toml" --cov=dask --cov-report=xml:"$WORKSPACE/dask-coverage.xml" --cov-report term
+rapids-logger "Python py.test for dask (LEGACY)"
+DASK_DATAFRAME__QUERY_PLANNING=False py.test $WORKSPACE -n 3 -v -m gpu --junitxml="$WORKSPACE/junit-dask-legacy.xml" --cov-config="$WORKSPACE/pyproject.toml" --cov=dask --cov-report=xml:"$WORKSPACE/dask-coverage-legacy.xml" --cov-report term
+
+rapids-logger "Python py.test for dask (NEW)"
+DASK_DATAFRAME__QUERY_PLANNING=True py.test $WORKSPACE -n 3 -v -m gpu --junitxml="$WORKSPACE/junit-dask.xml" --cov-config="$WORKSPACE/pyproject.toml" --cov=dask --cov-report=xml:"$WORKSPACE/dask-coverage.xml" --cov-report term
```

Comment:
@charlesbluca - How do you suggest we test both "legacy" and "dask-expr"? (We probably want to do this for 24.06, but can probably drop the legacy testing after that)

Reply:
The CI runtime with the entire test suite running twice is fast enough that it seems a little overkill to extend the CI matrix just to add a param for query planning that we'll probably remove shortly after - how do you feel about taking the approach you have here for now and punting that work off to if it becomes inconvenient or impossible to work with?

Also is it critical that we test the entire suite with query planning enabled, or could we get away with just testing some dataframe-specific modules?

Reply:
>Also is it critical that we test the entire suite with query planning enabled, or could we get away with just testing some dataframe-specific modules?

We should just test the `dataframe` module.